chapter_generative-adversarial-networks/dcgan.md

```diff
@@ -40,21 +59,45 @@ data_iter = gluon.data.DataLoader(
     shuffle=True, num_workers=d2l.get_dataloader_workers())
 ```
 
+```{.python .input}
+#@tab pytorch
+batch_size = 256
+transformer = torchvision.transforms.Compose([
+    torchvision.transforms.Resize((64, 64)),
+    torchvision.transforms.ToTensor(),
+    torchvision.transforms.Normalize(0.5, 0.5)
+])
+pokemon.transform = transformer
+data_iter = torch.utils.data.DataLoader(
+    pokemon, batch_size=batch_size,
+    shuffle=True, num_workers=d2l.get_dataloader_workers())
+```
+
 Let us visualize the first 20 images.
 
-```{.python .input  n=4}
+```{.python .input}
 d2l.set_figsize((4, 4))
 for X, y in data_iter:
     imgs = X[0:20,:,:,:].transpose(0, 2, 3, 1)/2+0.5
     d2l.show_images(imgs, num_rows=4, num_cols=5)
     break
 ```
 
+```{.python .input}
+#@tab pytorch
+warnings.filterwarnings('ignore')
+d2l.set_figsize((4, 4))
+for X, y in data_iter:
+    imgs = X[0:20,:,:,:].permute(0, 2, 3, 1)/2+0.5
+    d2l.show_images(imgs, num_rows=4, num_cols=5)
```

Comment:
Don't filter warnings. Solve the warning. Maybe don't use `permute` to achieve the same output without warnings.

Reply:
See my comments below.